src/validator.js

```diff
@@ -75,6 +88,44 @@ export default class Validator {
     return output;
   }
 
+  scan(_Xpi=Xpi) {
+    return new Promise((resolve, reject) => {
+      this.checkFileExists(this.packagePath)
+        .then(() => {
+          this.xpi = new _Xpi(this.packagePath);
+          return this.xpi.getMetaData();
+        }).then(() => {
+          return this.xpi.getJSFiles();
+        })
+        .then((jsFiles) => {
+          jsFiles.forEach((filename) => {
```

Comment:
Shouldn't this use Promise.all so you can resolve when all promises have resolved?